src/AppInstallerCommonCore/PathVariable.cpp

```diff
@@ -81,4 +95,10 @@ namespace AppInstaller::Registry::Environment
         std::wstring pathName = std::wstring{ s_PathName };
         m_key.SetValue(pathName, ConvertToUTF16(value), REG_EXPAND_SZ);
     }
+
+    void RefreshPathVariableForCurrentProcess()
+    {
+        const auto& pathValue = PathVariable(Manifest::ScopeEnum::User, true).GetPathValue() + PathVariable(Manifest::ScopeEnum::Machine, true).GetPathValue();
+        THROW_HR_IF(E_UNEXPECTED, _putenv_s("PATH", pathValue.c_str()) != 0);

```

Comment:
Use wide string version to support non ascii characters.

E_EXPECTED seems too strong here. Use a custom error code for this failure.

By the way, do we want to make this best effort? Or at least overridable with --force upon failure. Most of the cases, refresh env variable is not needed. Only a small portion of packages depend on refreshing dependency packages' newly written PATH.